File: 392. Is Subsequence/solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if not s:
            return True
        now = s[0]
        s = s[1:]
        for c in t:
            if c == now:
                if s == "":
                    return True
                else:
                    now = s[0]
                    s = s[1:]
        return False

File: 392. Is Subsequence/solution.go
package main

func isSubsequence(s string, t string) bool {
	if len(s) == 0 {
		return true
	}
	if len(s) > len(t) {
		return false
	}
	sIdx := 0
	for i := 0; i < len(t); i++ {
		if t[i] == s[sIdx] {
			sIdx++
		}
		if sIdx == len(s) {
			return true
		}
	}
	return sIdx == len(s)
}

func main() {
	isSubsequence("abc", "ahbgdc")
}
